vllm/v1/spec_decode/eagle.py

```diff
@@ -150,6 +178,15 @@ def propose(
             # [batch_size, 1]
             return draft_token_ids.view(-1, 1)
 
+        # TODO: Currently, MTP module released by deepseek only has
+        # one layer. Adapt this code to support multiple layers once
+        # there's a multi-layer MTP module.
+        if self.method == "deepseek_mtp":
+            logger.warning(
+                    "All Deepseek MTP models only have one layer. " \
+                    "Might need to change code to support multiple layers."
+                )
```

Comment:
Can we print this warning (only once) when initializing the `SpeculativeConfig`?
I think we shouldn't print this at the run time.